src/lambda-edge/parse-auth/index.ts

```diff
@@ -34,7 +34,14 @@ export const handler: CloudFrontRequestHandler = async (event) => {
             code_verifier: pkce
         });
 
-        const res = await httpPostWithRetry(`https://${cognitoAuthDomain}/oauth2/token`, body, { headers: { 'Content-Type': 'application/x-www-form-urlencoded' } });
+        let headers: { 'Content-Type': string, Authorization?: string } = { 'Content-Type': 'application/x-www-form-urlencoded' }
+
+        if(clientSecret !== '') {
```

Comment:
How about

```
if (clientSecret)
```
Or did I write too much python to appreciate that form

Reply:
I've been jumping between Python, Ruby, Typescript, and JS. Hard to remember what goes where anymore :)

Reply:
Updated